src/main/java/org/openstreetmap/atlas/checks/validation/points/InvalidMiniRoundaboutCheck.java

```diff
@@ -0,0 +1,143 @@
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.utilities.collections.Iterables;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * Flags any Nodes tagged as highway=MINI_ROUNDABOUT that do not have very many incoming/outgoing
+ * Edges as potentially tagged incorrectly.
+ *
+ * @author nachtm
+ */
+public class InvalidMiniRoundaboutCheck extends BaseCheck<Long>
+{
+    private static final long DEFAULT_VALENCE = 6;
+    private static final String MINIMUM_VALENCE_KEY = "valence.minimum";
+    private static final String OTHER_EDGES_INSTRUCTION = "This Mini-Roundabout Node ({0,number,#})"
+            + " has {1, number,#} connecting car-navigable Ways. Consider changing this.";
+    private static final String TWO_EDGES_INSTRUCTION = "This Mini-Roundabout Node ({0,number,#}) "
+            + "has 2 connecting car-navigable Ways. Consider changing this to highway=TURNING_LOOP or "
+            + "highway=TURNING_CIRCLE.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(TWO_EDGES_INSTRUCTION,
+            OTHER_EDGES_INSTRUCTION);
+    private static final DirectionTag[] VALID_DIRECTIONS = { DirectionTag.CLOCKWISE,
+            DirectionTag.ANTICLOCKWISE };
+    private final long minimumValence;
+
+    /**
+     * Construct an InvalidMiniRoundaboutCheck with the given configuration values.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public InvalidMiniRoundaboutCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.minimumValence = this.configurationValue(configuration, MINIMUM_VALENCE_KEY,
+                DEFAULT_VALENCE);
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Node
+                && Validators.isOfType(object, HighwayTag.class, HighwayTag.MINI_ROUNDABOUT);
+    }
+
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Node node = (Node) object;
+        final Collection<Edge> carNavigableEdges = this.getCarNavigableEdges(node);
+        final long valence = carNavigableEdges.size();
+        Optional<CheckFlag> result = Optional.empty();
+
+        // If this Node is a turnaround, we always want to flag it.
+        if (this.isTurnaround(carNavigableEdges))
+        {
+            result = Optional.of(this.flagNode(node, carNavigableEdges,
+                    this.getLocalizedInstruction(0, node.getOsmIdentifier())));
+        }
+        // Otherwise, if there is not a direction tag, and the valence is less than valenceMinimum,
+        // we want to flag it.
+        else if (!Validators.isOfType(node, DirectionTag.class, VALID_DIRECTIONS)
+                && valence < minimumValence && valence > 0)
+        {
+            result = Optional.of(this.flagNode(node, carNavigableEdges,
```

Comment:
So I believe this will flag if one of three ways coming out of the round-about is a one way, but not if they are all 2 way?
Is this intended? If not, I would prefer filtering for master edges in ```this.getCarNavigableEdges(node);``` , and then the valence is about the topology and view of what you see. 

This would simplify  ```this.isTurnaround(carNavigableEdges)``` as well to ```return carNavigableEdges.size() == 1```

Reply:
Yes, you are correct and this is the intended behavior. Often, nodes with a total valence of 5 are not actually mini-roundabouts (see [OSM ID 2367787523](https://www.openstreetmap.org/node/2367787523)). I can certainly add a comment to clarify this somewhere if you think it's unclear.

Reply:
no this is good. This was more confirming that it was as intended and thought about.